### ISSUE TITLE
Include clearer installation instructions

### ISSUE THREAD

**Issue by Alizter** [agda/cubical#12]
Currently the readme talks about making an agda sandbox, but doesn't specify what to do afterwards. For those who aren't so familiar with agda, this isn't completely obvious. Could you add some clearer installation instructions?

**Comment by Alizter** [comment]
Its not entirely clear how to compile and install the cubical library. The makefile talks about "agda" but it hasn't been fully installed in the sandbox anyway. I don't know how to do this in the agda sandbox, so I have run `make install` after making the sandbox, making the development version my main version of agda.

**Comment by Saizan** [comment]
For those not familiar with agda we could certainly have more detailed instructions.

If you don't mind, could you tell us if the following helps?

If you have agda master installed by following these instructions, given in the readme:
```
git clone https://github.com/agda/agda
cd agda
cabal sandbox init
cabal install
```

Then you want to "install" the emacs mode by running

```
agda-mode setup
```

Then you can just open the files from this repo in emacs and work with them. The emacs mode is briefly described here
https://agda.readthedocs.io/en/latest/getting-started/quick-guide.html#quick-guide-introduction


If you want to use cubical as a library in your own development you have to "register" its .agda-lib file as described here:
https://agda.readthedocs.io/en/latest/tools/package-system.html?highlight=library

**Comment by Alizter** [comment]
After running `cabal install` I cannot run `agda-mode setup`. There is no command, nor has agda been installed.

**Comment by Saizan** [comment]
`cabal install` by default installs things in a local directory, on linux it's `~/.cabal/bin`, maybe you do not have that in your `$PATH`.

**Comment by Alizter** [comment]
I've checked inside `~/.cabal/bin` there is no agda or agda-mode.

**Comment by Saizan** [comment]
I see, I don't use `cabal sandbox` myself, it seems you indeed have to copy the files from the sandbox to somewhere in your `$PATH`

https://stackoverflow.com/questions/27274475/cabal-install-an-executable-from-hackage-via-a-sandbox

@mortberg what do you do?

**Comment by mortberg** [comment]
I just update my PATH in `~/.bashrc`:
```
PATH=/home/mortberg/dev/agda/.cabal-sandbox/bin:$PATH
```

Writing some detailed install instructions in an INSTALL file sounds like a good idea. Maybe @Alizter can write a first draft once you get it to work? 

We should not forget to include some instructions for how to set it up as a library.

**Comment by Alizter** [comment]
I have managed to get it to work now. I installed agda-2.6.0 without using the sandbox.

A big potential confusion is definitely the $PATH variable. We will just have to tell users to make sure they have `$HOME/.cabal/bin` added to their path in `.basrc`. (And maybe restart the terminal so it reloads `.bashrc`).

Now I understand why the sandbox is used. If you don't have agda then there is no problem, do what I did. But if you already do then the library doesn't simply add on (yet). The current agda version is agda-2.5 yet this library only works for agda-2.6 and upwards.

A possible solution to this is having an install script? It would need to make sure there is the correct version of agda-2.6, if not install it in a sandbox somewhere, compile it and move the executable to the users bin (along with agda-mode). Then the sandbox can be rid of. (However I am not sure how this will interact with existing versions of agda).
